juju/juju
# Where possible write fstab entries using the device UUID

## Please provide the following details to expedite Pull Request review:

## Description of change

Juju creates an fstab entry for each attached volume mount. It was using device path but this PR changed to use UUID instead.
Depending on the cloud, a volume may be mounted as either a disk or a partition. So we add logic to handle the partition appearing with a UUID after we may have already noticed the disk. The fstab updating will correctly replace any device path entry with the UUID one.
A nofail option is also added to the fstab emtry.

## QA steps

On aws, google, and azure, bootstrap and deploy a charm with storage
(aws using disks and azure and google use partitions so check all 3)
eg for aws
`juju deploy postgresql --storage pgdata=ebs,1G`

ssh into the machine and look at /etc/fstab, there should be a UUID mount, eg
```
# /srv/pgdata was on /dev/sdb1 during installation
UUID=2a3a8eca-b9bd-4c53-942d-085d14a56222 /srv/pgdata ext4 nofail,relatime,rw 0 0
```

reboot the machine and check the mount is recreated after.


## Bug reference

https://bugs.launchpad.net/juju/+bug/1877380
